### DISCUSSION TITLE
Only allow images on files field

### DISCUSSION THREAD

**Discussion by oberbayer** [directus/directus#14457] [Archive]
May be a noob question, but I've tried validation rules like "Files Field Name -> Directus Files ID -> Mime Type Equals image/jpeg" to only allow JPG-images to be uploaded via the multiple files field (my aim here is to build a gallery). But whatever I do (equals, not doesn't equal, begins with etc.) - I am never allowed to save my entry.
Moreover, the files are already uploaded to the system, even if I am not able to save the entry.

Is this a bug or am I doing anything wrong? And is there a way to check for mime type already during upload?

Thank you for any hint on how to build galleries!

**Comment by rijkvanzanten** [comment]
There's currently no way to configure the validation rules to check against a mime type

**Reply by oberbayer** [reply]
Okay. Thank you, Rijk.

**Reply by oberbayer** [reply]
But do you have a tip for me on how to build galleries? Check against mime type in the frontend?

**Reply by JohnHillegass** [reply]
@rijkvanzanten any suggestions on where a config like this could be added? I was thinking about something similar around a filter hook but realized it would be nice to configure in the project settings.

**Reply by martin-braun** [reply]
There is a `type` field now, is this the right thing? @rijkvanzanten 

If so, how can I make it only accept JPG and PNG, please?

**Comment by Tommy1223454** [comment]
This is the exact failure mode I try to avoid in upload design.

If the validation result arrives only after the file is already part of the system, then the “acceptance boundary” is happening too late.

I built a Node.js OSS package for moving that decision earlier:
https://github.com/pompelmi/pompelmi

The pattern I like is parse -> inspect -> then persist/promote, rather than uploading first and rejecting later at the form/document layer.
